### PR TITLE
Pre for forms and passwords.

### PR DESCRIPTION
**Pull request by ncalexan**

### CODE REVIEW

**Comment by liuche** [comment]
Looks good to me, except for needing to rebase.

File: src/main/java/org/mozilla/gecko/sync/repositories/android/AndroidBrowserRepositoryDataAccessor.java [hunk 1]
```diff
@@ -61,9 +61,20 @@ public AndroidBrowserRepositoryDataAccessor(Context context) {
   }
 
   protected abstract String[] getAllColumns();
+
+  /**
+   * Convert a <code>Record</code> into a <code>ContentValues</code>.
```
Comment (rnewman) [inline src/main/java/org/mozilla/gecko/sync/repositories/android/AndroidBrowserRepositoryDataAccessor.java]: "Produce a ContentValues instance that represents the provided Record".

File: src/main/java/org/mozilla/gecko/sync/repositories/android/AndroidBrowserHistoryDataAccessor.java [hunk 2]
```diff
@@ -115,10 +115,10 @@ public void update(String oldGUID, Record newRecord) {
   }
 
   @Override
-  protected void delete(String guid) {
-    Log.d(LOG_TAG, "Deleting record " + guid);
-    super.delete(guid);
+  public int purgeGuid(String guid) {
+    Log.d(LOG_TAG, "Purging record with " + guid);
```
Comment (rnewman) [inline src/main/java/org/mozilla/gecko/sync/repositories/android/AndroidBrowserHistoryDataAccessor.java]: Logger!

File: src/main/java/org/mozilla/gecko/sync/repositories/android/AndroidBrowserRepositoryDataAccessor.java [hunk 3]
```diff
@@ -86,33 +97,31 @@ public void wipe() {
...
-      }
-      while (!cur.isAfterLast()) {
-        delete(RepoUtils.getStringFromCursor(cur, BrowserContract.SyncColumns.GUID));
-        cur.moveToNext();
-      }
-    } finally {
-      cur.close();
-    }
+    Uri uri = getUri();
+    Logger.info(LOG_TAG, "Purging deleted from: " + uri);
```
Comment (rnewman) [inline src/main/java/org/mozilla/gecko/sync/repositories/android/AndroidBrowserRepositoryDataAccessor.java:104]: debug.

**Comment by rnewman** [comment]
Three teensy tiny nits. If tests pass, fix and merge.

**Pull request merged**